lib/scholar/preprocessing.ex

```diff
@@ -87,4 +87,78 @@ defmodule Scholar.Preprocessing do
     opts = keyword!(opts, threshold: 0, type: {:f, 32})
     (tensor > opts[:threshold]) |> Nx.as_type(opts[:type])
   end
+
+  @doc """
+  Encodes a tensor's values into integers from range 0 to `:num_classes - 1`.
+
+  ## Options
+
+    * `:num_classes` - Number of classes to be encoded. Required.
+
+  ## Examples
+
+      iex> Scholar.Preprocessing.ordinal_encoding(Nx.tensor([3, 2, 4, 56, 2, 4, 2]), num_classes: 4)
+      #Nx.Tensor<
+        s64[7]
+        [1, 0, 2, 3, 0, 2, 0]
+      >
+  """
+  @spec ordinal_encoding(tensor :: Nx.Tensor.t(), opts :: Keyword.t()) :: Nx.Tensor.t()
+  defn ordinal_encoding(tensor, opts \\ []) do
```

Comment:
@josevalim @polvalente @seanmor5 any preferences on the naming? Perhaps `encode_oridinal`/`encode_one_hot`?

Reply:
`one_hot_encode`

Reply:
We could also call it `one_hot_encoder`, which would match `standard_scaler`. But perhaps it should be `one_hot_encode` and `standard_scale`.

Reply:
+1 for the `one_hot_encode` and `standard_scale`.

Reply:
@msluszniak let's go with one_hot_encode then! I will open up an issue to rename the other.